FuelLabs/fuel-indexer
# chore: create fixture for Fuel node

## Changelog
- Add fixture for Fuel node and adjusts E2E tests to use it
- Remove `fuel-node` package from workspace
- Remove `fuel-node` build step from CI
- Remove `e2e.bash` test script
- Adjust E2E tests to remove unnecessary asserts
- Add small `hello-world-test-node` package that wraps the aforementioned test fixture
- Adjusts documentation for Hello World example

## Testing Steps
CI should pass with new Fuel node fixture. Additionally, the "Hello World" example has been adjusted; run through both the example README and the example in the documentation, and verify that everything works.

## Notes
- No big improvement in CI runtime; however, it does allow for all of our testing to be done through `cargo` actions and for us not to use extra scripts.
- A number of tests needed to be adjusted to either remove asserts that weren't properly testing values or to delete records from tables at test invocation to ensure a clean slate.
